Dockerfile

```diff
@@ -1,14 +1,40 @@
-FROM gradle:jdk8 as builder
+FROM gradle:jdk8 as cache
+RUN mkdir -p /home/gradle/cache
+ENV GRADLE_USER_HOME /home/gradle/cache
+COPY build.gradle /home/gradle/src/
+WORKDIR /home/gradle/src
+RUN gradle clean dependencies -i --stacktrace
 
+FROM gradle:jdk8 AS builder
+COPY --from=cache /home/gradle/cache /home/gradle/.gradle
 COPY --chown=gradle:gradle . /home/gradle/src
+
 WORKDIR /home/gradle/src
-RUN set https_proxy=https://127.0.0.1:8001
-RUN set http_proxy=https://127.0.0.1:8001
-RUN gradle build
+RUN gradle clean build
+RUN mkdir  -p target
```

Comment:
Two space `  ` exists, and `-p` is not necessary in this situation.
Should be `RUN mkdir target`